From 63fe9a643018f30c85e21d2202c4f0c0667ac2bb Mon Sep 17 00:00:00 2001
From: Paul Lemire <paul.lemire@kdab.com>
Date: Wed, 3 Mar 2021 12:08:51 +0100
Subject: [PATCH 21/33] Restore ObjectLevelLockingPolicy on ShaderManager

We are hitting a race condition otherwise. This is likely due to the fact
that removing the shaders is done only once we have unloaded them from the
graphics API which technically could happen in a separate thread than the
scenegraph thread when running with Scene3D.

Pick-to: 6.0 6.1 5.15
Change-Id: Ic914f3b305bf807081294a90250413a5d263f124
Reviewed-by: Mike Krus <mike.krus@kdab.com>
(cherry picked from commit a536039d102f911677e5efec03cc6f3d701aceac)
---
 src/render/backend/managers_p.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/render/backend/managers_p.h b/src/render/backend/managers_p.h
index a3d42d24a..864c168be 100644
--- a/src/render/backend/managers_p.h
+++ b/src/render/backend/managers_p.h
@@ -200,7 +200,7 @@ public:
 class Q_3DRENDERSHARED_PRIVATE_EXPORT ShaderManager : public Qt3DCore::QResourceManager<
         Shader,
         Qt3DCore::QNodeId,
-        Qt3DCore::NonLockingPolicy>
+        Qt3DCore::ObjectLevelLockingPolicy>
 {
 public:
     ShaderManager() {}
-- 
2.33.0

